Gems/Prefab/PrefabDependencyViewer/Code/Source/PrefabDependencyViewerEditorSystemComponent.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzToolsFramework/API/ViewPaneOptions.h>
+#include <PrefabDependencyViewerEditorSystemComponent.h>
+#include <MainWindow.h>
+#include <QMenu>
+#include <LyViewPaneNames.h>
+#include <PrefabDependencyTree.h>
+
+namespace PrefabDependencyViewer
+{
+    using Outcome = AZ::Outcome<PrefabDependencyTree, const char*>;
+
+    InstanceEntityMapperInterface* PrefabDependencyViewerEditorSystemComponent::s_prefabEntityMapperInterface = nullptr;
+    PrefabSystemComponentInterface* PrefabDependencyViewerEditorSystemComponent::s_prefabSystemComponentInterface = nullptr;
+    PrefabPublicInterface* PrefabDependencyViewerEditorSystemComponent::s_prefabPublicInterface = nullptr;
+
+    const char* PrefabDependencyViewerEditorSystemComponent::s_prefabViewerTitle = "Prefab Dependencies Viewer";
+
+    void PrefabDependencyViewerEditorSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<PrefabDependencyViewerEditorSystemComponent, AZ::Component>()
+                ->Version(0);
+        }
+    }
+
+    PrefabDependencyViewerEditorSystemComponent::PrefabDependencyViewerEditorSystemComponent() = default;
+
+    void PrefabDependencyViewerEditorSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("PrefabDependencyViewerEditorService"));
+    }
+
+    void PrefabDependencyViewerEditorSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("PrefabDependencyViewerEditorService"));
+    }
+
+    void PrefabDependencyViewerEditorSystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+    }
+
+    void PrefabDependencyViewerEditorSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
```

Comment:
GetRequiredServices should probably note that it depends on the prefab system component
So you'd add
```
required.push_back(AZ_CRC_CE("PrefabSystem"));
```

This will indicate that the PrefabSystemComponent will be initialized before this component and shut down after it. (Since the PrefabSystemComponent publishes that it "Provides" the PrefabSystem "service"

See the comments in Code/Framework/AzCore/AzCore/Component/Component.h for explanation of GetRequired vs GetDependent

Reply:
I think I want PrefabDependencyViewerSystemComponent to return an error and not get activated if I can't get PrefabSystemComponent services. So I'm going to use GetRequiredServices method.

The reason behind that is I need PrefabSystemComponentInterface in order to find TemplateDoms which in turn help create the DependencyTree. => No need to activate the component if it can't do it's job.

Resolved

Reply:
Just to make sure this is the behavior you want:
GetRequiredServices - cause the app to not startup if its missing this.  The code doesn't have to deal with the service missing.
GetDependentServices - do not cause the app to not start.  App can start even if they're missing.  Rest of code has to gracefully handle missing service.

This means if someone loads this module into a different tool like Material Editor it will stop Material Editor from working.  They shouldn't load it there of course, so this isn't something you need to fix.  But its a choice to make.